src/bindgen/ir/generic_path.rs

```diff
@@ -298,6 +348,9 @@ impl GenericPath {
             _ => Vec::new(),
         };
 
-        Ok(Self::new(path, generics))
+        let mut ret = Self::new(path, generics);
+        ret.assoc = assoc;
```

Comment:
```suggestion
        let ret = Self::new(path, qself, generics);
```

I would prefer you do it inside `Self::new`

Reply:
Do you mean to add the AssocTypeID as a parameter in the GenericPath::new()? The GenericPath::new() is called in many other places and they would mostly result in changing the call from "GenericPath::new(path, generics)" to "GenericPath::new(path, None, generics)" 

The new function signature will look like this:
```pub fn new(path: Path, assoc: Option<AssocTypeId>, generics: Vec<GenericArgument>)```

Reply:
> The new function signature will look like this: `pub fn new(path: Path, assoc: Option<AssocTypeId>, generics: Vec<GenericArgument>)`

no, I expected it would look like this: `pub fn new(path: Path, qself: &Option<syn::QSelf>, generics: Vec<GenericArgument>)`

Reply:
The path in Genericpath::new() is an "ir::path::Path" defined within cbindgen (it's essentially a string). On the other hand, the path I used to generate the AssocTypeId::new() is a syn::Path. I need the syn::Path in order to parse the name of the trait, in the cbindgen's ir path, they omit some important info by only keeping the last segment.

Reply:
ok, let's wait with this one

Reply:
here is what we're going to do. You will change function signature into `pub fn new(path: Path, qself: &Option<Type>, generics: Vec<GenericArgument>)`

Reply:
Trying to understand this. So does this mean I will pass a &Option<Type> to GenericPath::new() and then from inside the GenericPath::new() function, I call AssocTypeId to generate the ID? AssocTypeID::new() currently takes a syn::Path and syn::QSelf in order to generate the ID. If I only pass it the already parsed Type, it won't have information about the trait and associated type name.

Reply:
Notice that functions which are named `new` take cbindgen types, while functions named `load` take `syn` types

Reply:
So I'm saying make the function `new` take `cbindgen` types. Any parsing that needs to be done to obtain `cbindgen` types should be done outside `new`